buildkite/buildkite.go

```diff
@@ -99,12 +91,12 @@ func (c *Client) GetAgentMetrics(queue string) (AgentMetrics, error) {
 	return metrics, nil
 }
 
-func (c *Client) queryMetrics(into interface{}) (pollDuration time.Duration, err error) {
+func (c *Client) queryMetrics(into interface{}, queue string) (pollDuration time.Duration, err error) {
 	endpoint, err := url.Parse(c.Endpoint)
 	if err != nil {
 		return time.Duration(0), err
 	}
-	endpoint.Path += "/metrics"
+	endpoint.Path += fmt.Sprintf("/metrics/queue?name=%s", queue)
```

Comment:
Probably safer to pass params through the right field, encoded:

```suggestion
	endpoint.Path += "/metrics/queue"
	q := url.Values{"name": []string{queue}}
	endpoint.RawQuery = q.Encode()
```